<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>VirtualTrackpad</class>
 <widget class="QWidget" name="VirtualTrackpad">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1247</width>
    <height>748</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <layout class="QGridLayout" name="gridLayout_5">
   <item row="0" column="0">
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QWidget" name="widget_disconnect" native="true">
       <layout class="QGridLayout" name="gridLayout">
        <item row="0" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <spacer name="horizontalSpacer_2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <spacer name="verticalSpacer">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout">
              <item>
               <widget class="QLabel" name="label">
                <property name="minimumSize">
                 <size>
                  <width>96</width>
                  <height>96</height>
                 </size>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>96</width>
                  <height>96</height>
                 </size>
                </property>
                <property name="text">
                 <string/>
                </property>
                <property name="pixmap">
                 <pixmap resource="../../Res/UI.qrc">:/UI/Image/TrackpadDisconnected.png</pixmap>
                </property>
                <property name="scaledContents">
                 <bool>true</bool>
                </property>
                <property name="alignment">
                 <set>Qt::AlignCenter</set>
                </property>
               </widget>
              </item>
              <item>
               <spacer name="horizontalSpacer">
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
                <property name="sizeType">
                 <enum>QSizePolicy::Fixed</enum>
                </property>
                <property name="sizeHint" stdset="0">
                 <size>
                  <width>20</width>
                  <height>20</height>
                 </size>
                </property>
               </spacer>
              </item>
              <item>
               <layout class="QVBoxLayout" name="verticalLayout">
                <item>
                 <widget class="ElaText" name="label_2">
                  <property name="minimumSize">
                   <size>
                    <width>400</width>
                    <height>0</height>
                   </size>
                  </property>
                  <property name="text">
                   <string>Robot is not connected</string>
                  </property>
                  <property name="alignment">
                   <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="ElaText" name="label_subtitle">
                  <property name="font">
                   <font>
                    <pointsize>10</pointsize>
                   </font>
                  </property>
                  <property name="text">
                   <string>Robot is not connected, please connect to robot first before using virtual trackpad.</string>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </item>
          <item>
           <spacer name="horizontalSpacer_3">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QWidget" name="widget_connected" native="true">
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="0">
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="SimpleTrackpad" name="widget_LeftTrack" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>280</width>
              <height>270</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>280</width>
              <height>270</height>
             </size>
            </property>
            <property name="autoFillBackground">
             <bool>true</bool>
            </property>
            <layout class="QGridLayout" name="gridLayout_3"/>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_4">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>28</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QWidget" name="widget_buttons" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <layout class="QGridLayout" name="gridLayout_6">
             <item row="0" column="0">
              <layout class="QVBoxLayout" name="verticalLayout_6">
               <item>
                <layout class="QHBoxLayout" name="horizontalLayout_4">
                 <item>
                  <widget class="ElaText" name="label_buttons">
                   <property name="font">
                    <font>
                     <pointsize>14</pointsize>
                    </font>
                   </property>
                   <property name="text">
                    <string>Event Keyboard</string>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="horizontalSpacer_6">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>40</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                 <item>
                  <widget class="ElaToggleButton" name="ToggleButtonModeSwitch" native="true">
                   <property name="sizePolicy">
                    <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                     <horstretch>0</horstretch>
                     <verstretch>0</verstretch>
                    </sizepolicy>
                   </property>
                   <property name="minimumSize">
                    <size>
                     <width>130</width>
                     <height>30</height>
                    </size>
                   </property>
                   <property name="maximumSize">
                    <size>
                     <width>130</width>
                     <height>30</height>
                    </size>
                   </property>
                  </widget>
                 </item>
                 <item>
                  <spacer name="horizontalSpacer_5">
                   <property name="orientation">
                    <enum>Qt::Horizontal</enum>
                   </property>
                   <property name="sizeType">
                    <enum>QSizePolicy::Fixed</enum>
                   </property>
                   <property name="sizeHint" stdset="0">
                    <size>
                     <width>10</width>
                     <height>20</height>
                    </size>
                   </property>
                  </spacer>
                 </item>
                </layout>
               </item>
               <item>
                <widget class="QWidget" name="ValueModeWidget" native="true">
                 <layout class="QGridLayout" name="gridLayout_8">
                  <item row="0" column="0">
                   <layout class="QHBoxLayout" name="horizontalLayout_6">
                    <item>
                     <spacer name="horizontalSpacer_8">
                      <property name="orientation">
                       <enum>Qt::Horizontal</enum>
                      </property>
                      <property name="sizeHint" stdset="0">
                       <size>
                        <width>40</width>
                        <height>20</height>
                       </size>
                      </property>
                     </spacer>
                    </item>
                    <item>
                     <layout class="QHBoxLayout" name="horizontalLayout_7">
                      <item>
                       <layout class="QVBoxLayout" name="verticalLayout_3">
                        <item>
                         <widget class="ElaText" name="label_valuetype">
                          <property name="text">
                           <string>Value Type</string>
                          </property>
                         </widget>
                        </item>
                        <item>
                         <widget class="ElaRadioButton" name="radioButton_int">
                          <property name="text">
                           <string>Int</string>
                          </property>
                          <property name="checked">
                           <bool>true</bool>
                          </property>
                         </widget>
                        </item>
                        <item>
                         <widget class="ElaRadioButton" name="radioButton_double">
                          <property name="text">
                           <string>Double</string>
                          </property>
                         </widget>
                        </item>
                       </layout>
                      </item>
                      <item>
                       <spacer name="horizontalSpacer_7">
                        <property name="orientation">
                         <enum>Qt::Horizontal</enum>
                        </property>
                        <property name="sizeType">
                         <enum>QSizePolicy::Fixed</enum>
                        </property>
                        <property name="sizeHint" stdset="0">
                         <size>
                          <width>40</width>
                          <height>20</height>
                         </size>
                        </property>
                       </spacer>
                      </item>
                      <item>
                       <layout class="QVBoxLayout" name="verticalLayout_4">
                        <item>
                         <layout class="QHBoxLayout" name="horizontalLayout_8">
                          <item>
                           <widget class="ElaText" name="label_keyname">
                            <property name="text">
                             <string>Button Name:  </string>
                            </property>
                           </widget>
                          </item>
                          <item>
                           <widget class="ElaComboBox" name="comboBox_keyName">
                            <property name="sizePolicy">
                             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                              <horstretch>0</horstretch>
                              <verstretch>0</verstretch>
                             </sizepolicy>
                            </property>
                            <property name="minimumSize">
                             <size>
                              <width>156</width>
                              <height>32</height>
                             </size>
                            </property>
                            <property name="maximumSize">
                             <size>
                              <width>156</width>
                              <height>32</height>
                             </size>
                            </property>
                           </widget>
                          </item>
                         </layout>
                        </item>
                        <item>
                         <layout class="QHBoxLayout" name="horizontalLayout_9">
                          <item>
                           <widget class="ElaText" name="label_keyvalue">
                            <property name="text">
                             <string>Button Value:  </string>
                            </property>
                           </widget>
                          </item>
                          <item>
                           <widget class="ElaLineEdit" name="lineEdit_keyValue">
                            <property name="sizePolicy">
                             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
                              <horstretch>0</horstretch>
                              <verstretch>0</verstretch>
                             </sizepolicy>
                            </property>
                            <property name="minimumSize">
                             <size>
                              <width>150</width>
                              <height>30</height>
                             </size>
                            </property>
                            <property name="maximumSize">
                             <size>
                              <width>150</width>
                              <height>30</height>
                             </size>
                            </property>
                           </widget>
                          </item>
                          <item>
                           <spacer name="horizontalSpacer_10">
                            <property name="orientation">
                             <enum>Qt::Horizontal</enum>
                            </property>
                            <property name="sizeType">
                             <enum>QSizePolicy::Fixed</enum>
                            </property>
                            <property name="sizeHint" stdset="0">
                             <size>
                              <width>2</width>
                              <height>20</height>
                             </size>
                            </property>
                           </spacer>
                          </item>
                         </layout>
                        </item>
                       </layout>
                      </item>
                      <item>
                       <widget class="ElaPushButton" name="pushButton_buttonApply">
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                          <horstretch>0</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="minimumSize">
                         <size>
                          <width>100</width>
                          <height>28</height>
                         </size>
                        </property>
                        <property name="text">
                         <string>Apply</string>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </item>
                    <item>
                     <spacer name="horizontalSpacer_9">
                      <property name="orientation">
                       <enum>Qt::Horizontal</enum>
                      </property>
                      <property name="sizeHint" stdset="0">
                       <size>
                        <width>40</width>
                        <height>20</height>
                       </size>
                      </property>
                     </spacer>
                    </item>
                   </layout>
                  </item>
                 </layout>
                </widget>
               </item>
               <item>
                <widget class="ElaScrollArea" name="ButtonModeScrollArea">
                 <property name="palette">
                  <palette>
                   <active>
                    <colorrole role="Base">
                     <brush brushstyle="SolidPattern">
                      <color alpha="0">
                       <red>255</red>
                       <green>255</green>
                       <blue>255</blue>
                      </color>
                     </brush>
                    </colorrole>
                    <colorrole role="Window">
                     <brush brushstyle="SolidPattern">
                      <color alpha="0">
                       <red>240</red>
                       <green>240</green>
                       <blue>240</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </active>
                   <inactive>
                    <colorrole role="Base">
                     <brush brushstyle="SolidPattern">
                      <color alpha="0">
                       <red>255</red>
                       <green>255</green>
                       <blue>255</blue>
                      </color>
                     </brush>
                    </colorrole>
                    <colorrole role="Window">
                     <brush brushstyle="SolidPattern">
                      <color alpha="0">
                       <red>240</red>
                       <green>240</green>
                       <blue>240</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </inactive>
                   <disabled>
                    <colorrole role="Base">
                     <brush brushstyle="SolidPattern">
                      <color alpha="0">
                       <red>240</red>
                       <green>240</green>
                       <blue>240</blue>
                      </color>
                     </brush>
                    </colorrole>
                    <colorrole role="Window">
                     <brush brushstyle="SolidPattern">
                      <color alpha="0">
                       <red>240</red>
                       <green>240</green>
                       <blue>240</blue>
                      </color>
                     </brush>
                    </colorrole>
                   </disabled>
                  </palette>
                 </property>
                 <property name="autoFillBackground">
                  <bool>true</bool>
                 </property>
                 <property name="widgetResizable">
                  <bool>true</bool>
                 </property>
                 <widget class="QWidget" name="scrollAreaWidgetContents">
                  <property name="geometry">
                   <rect>
                    <x>0</x>
                    <y>0</y>
                    <width>579</width>
                    <height>179</height>
                   </rect>
                  </property>
                 </widget>
                </widget>
               </item>
              </layout>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="SimpleTrackpad" name="widget_RightTrack" native="true">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>280</width>
              <height>270</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>280</width>
              <height>270</height>
             </size>
            </property>
            <property name="autoFillBackground">
             <bool>true</bool>
            </property>
            <layout class="QGridLayout" name="gridLayout_2"/>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ElaText</class>
   <extends>QLabel</extends>
   <header>ElaText.h</header>
  </customwidget>
  <customwidget>
   <class>ElaScrollArea</class>
   <extends>QScrollArea</extends>
   <header>ElaScrollArea.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>SimpleTrackpad</class>
   <extends>QWidget</extends>
   <header>SimpleTrackpad.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ElaPushButton</class>
   <extends>QPushButton</extends>
   <header>ElaPushButton.h</header>
  </customwidget>
  <customwidget>
   <class>ElaComboBox</class>
   <extends>QComboBox</extends>
   <header>ElaComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>ElaLineEdit</class>
   <extends>QLineEdit</extends>
   <header>ElaLineEdit.h</header>
  </customwidget>
  <customwidget>
   <class>ElaToggleButton</class>
   <extends>QWidget</extends>
   <header>ElaToggleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ElaRadioButton</class>
   <extends>QRadioButton</extends>
   <header>ElaRadioButton.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../../Res/UI.qrc"/>
 </resources>
 <connections/>
</ui>
